### PR TITLE
Remove hardcoding of init path based on node type

### PR DESCRIPTION
**Pull request by jbowen93**
Fix dockerfile so that it doesn't hardcode the init path.

The init path is still variable but now dependent on the logic in the `celestia-node` code itself and not the `entrypoint.sh`.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,5 @@
 FROM --platform=$BUILDPLATFORM golang:1.18 as builder
-RUN apt-get update && \
-    apt-get install make bash
```
Comment (Wondertan) [inline docker/Dockerfile:3]: Why do we remove 'update' and bash installation?

**Review by Wondertan — APPROVED** [review]
LGTM

File: docker/Dockerfile (hunk 1 above)
Comment (jbowen93) [reply docker/Dockerfile:3]: Because this is the builder image and these aren't needed to perform the build successfully. Just hygiene.

**Review by adlerjohn — APPROVED** [review]
utACK

**Review by renaynay — APPROVED** [review]

**Pull request merged**